### PR TITLE
Preload everything using Background Process library

### PR DESCRIPTION
**Pull request by Tabrisrp**

### CODE REVIEW

File: inc/classes/class-rocket-partial-background-preload.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	 *
+	 * @param mixed $item Queue item to iterate over.
+	 *
+	 * @return bool true if exists, false otherwise
+	 */
+	protected function is_already_cached( $item ) {
+		$host = ( isset( $_SERVER['HTTP_HOST'] ) ) ? $_SERVER['HTTP_HOST'] : time();
+		$host = trim( strtolower( $host ), '.' );
+		$host = urlencode( $host );
+		$host = isset( $rocket_url_no_dots ) ? str_replace( '.', '_', $host ) : $host;
```
Comment (Screenfeed) [inline inc/classes/class-rocket-partial-background-preload.php]: Where does `$rocket_url_no_dots` come from?

File: inc/classes/class-rocket-sitemap-background-preload.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	/**
+	 * Check if the cache file for $item already exists
+	 *
+	 * @param mixed $item Queue item to iterate over.
+	 *
+	 * @return bool true if exists, false otherwise
+	 */
+	protected function is_already_cached( $item ) {
+		$url = wp_parse_url( $item );
+		$host = isset( $url['host'] ) ? str_replace( '.', '_', $url['host'] ) : $url['host'];
```
Comment (Screenfeed) [inline inc/classes/class-rocket-sitemap-background-preload.php]: So if `$url['host']` is not set, you use it anyway?

File: inc/classes/class-rocket-sitemap-background-preload.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		$file_cache_path = WP_ROCKET_CACHE_PATH . $host . '/' . strtolower( $url['path'] ) . '/index.html';
```
Comment (Screenfeed) [inline inc/classes/class-rocket-sitemap-background-preload.php]: What if `$url['path']` is not set?
If empty or equal `/`, will it be a problem?

File: inc/deprecated.php [hunk 4]
```diff
@@ -374,3 +374,53 @@ function get_rocket_ecommerce_exclude_pages() {
 		return array();
 	}
 endif;
+
+if ( ! function_exists( 'run_rocket_bot' ) ) :
```
Comment (Screenfeed) [inline inc/deprecated.php]: Why use `function_exists()` on your deprecated functions?

File: inc/functions/preload.php [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+		 *
+		 * @since 2.8
+		 *
+		 * @param string $sitemap_type 	the sitemap identifier
+		 * @param string $sitemap_url sitemap URL crawled
+		*/
+		do_action( 'after_run_rocket_sitemap_preload', $sitemap_type, $sitemap_url );
+	}
+
+	foreach ( $urls_group as $k => $urls ) {
```
Comment (Screenfeed) [inline inc/functions/preload.php]: `$k` is useless.

File: inc/functions/preload.php [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	$args = apply_filters( 'rocket_preload_sitemap_request_args', 
+		array(
+			'timeout'    => 0.01,
+			'blocking'   => false,
+			'user-agent' => 'wprocket-sitemap-preload',
+			'sslverify'  => apply_filters( 'https_local_ssl_verify', true ),
+		)
+	);
+
+	$sitemap = wp_remote_get( esc_url_raw( $sitemap_url ) );
```
Comment (Screenfeed) [inline inc/functions/preload.php]: It should be `esc_url()` I think. We generally use `esc_url_raw()` when redirecting or storing in DB.

File: inc/classes/class-rocket-partial-background-preload.php (hunk 1 above)
Comment (Tabrisrp) [reply inc/classes/class-rocket-partial-background-preload.php]: it's defined in the rocket config file in `wp-content/wp-rocket-config/`

see https://github.com/wp-media/wp-rocket/blob/master/inc/functions/files.php#L167

File: inc/deprecated.php (hunk 4 above)
Comment (Tabrisrp) [reply inc/deprecated.php]: I followed the pattern used for this file, if it's useless I can remove them

File: inc/classes/class-rocket-partial-background-preload.php (hunk 1 above)
Comment (Screenfeed) [reply inc/classes/class-rocket-partial-background-preload.php]: OK but how can you access it from within this method? I don't see any `global $rocket_url_no_dots;` or include :|

File: inc/deprecated.php (hunk 4 above)
Comment (Screenfeed) [reply inc/deprecated.php]: I can't see in which cases it could be useful. But anyway, don't bother, it's not a problem, I was just curious.

**Comment by Tabrisrp** [comment]
PR #1222 is the new version of this

**Pull request closed without merging**